### PR TITLE
Only set 'syn-sync' when it is the main syntax

### PR DESCRIPTION
**Pull request by pjio**

### CODE REVIEW

**Comment by pjio** [comment]
This PR tries another approach and sets a flag in the files which include the nested syntax.

I tested with `nvim v0.4.3` and disabled my other plugins:

`:syn sync` shows for graphql files as intended:
`syncing starts 500 lines before top line`

For php files `:syn sync` show the original:
`syncing starts 2147483647 lines before top line`

I also tested that the nested syntax is still working. Very nice feature to have!

File: after/syntax/javascript/graphql.vim [hunk 1]
```diff
@@ -25,7 +25,11 @@ if exists('b:current_syntax')
   let s:current_syntax = b:current_syntax
   unlet b:current_syntax
 endif
+
+let g:graphql_nested_syntax = 1
```
Comment (jparise) [inline after/syntax/javascript/graphql.vim]: I think these could be buffer-local values (`b:graphql_nested_syntax`).

Comment (pjio) [reply after/syntax/javascript/graphql.vim]: Good point, I updated the commit and it's still working as expected!

File: syntax/graphql.vim [hunk 2]
```diff
@@ -90,6 +90,8 @@ hi def link graphqlStructure        Structure
 hi def link graphqlType             Type
 hi def link graphqlVariable         Identifier
 
-syn sync minlines=500
+if !exists('b:graphql_nested_syntax') || b:graphql_nested_syntax == 0
```
Comment (jparise) [inline syntax/graphql.vim]: I think this can also be simplified:
```suggestion
if !get(b:, 'graphql_nested_syntax')
```

**Review by jparise — COMMENTED** [review]
Do you think there's a good way to write a test for this behavior to prevent future regressions?

**Comment by pjio** [comment]
I changed the condition to `if !get(b:, 'graphql_nested_syntax')`

To write a unit test is more tricky than I first thought. The problem is to set `minlines` to a predictable value **before** the buffer is loaded.

I had some success with adding this to `test/php/default.vader`
```
Execute (Settings assertions):
  redir @a | syntax sync | redir END
  AssertEqual 'syncing starts 2147483647 lines before top line', @a
```
But on the CI there are different defaults:
image

For now I think it's not easily possible.

**Comment by pjio** [comment]
I added a test which checks for the '500' in the syntax setting.

**Comment by pjio** [comment]
The test checks now for the exact value which would be set if there is a regression. The downside is, the test has to be changed whenever the value "500" is changed or the string format changes. The upside is, it shouldn't fail as a false positive. I had to store the result of `match()` in a local variable because `Assert` didn't handle the comparison as a boolean expression (I assume a limitation of Vader..)

**Review by jparise — APPROVED** [review]
This looks good! I'll give it a final look over the weekend and then merge.

**Comment by jparise** [comment]
Thanks for this, @pjio! I appreciate you digging in to the issue and producing a generalized solution to the problem.

**Pull request merged**